TST: ASDF 2.5.0 broke some tests

Our CI is now picking up [ASDF 2.5.0](https://github.com/spacetelescope/asdf/releases/tag/2.5.0), which broke the following tests:

```
____________________________ test_backwards_compat _____________________________
    def test_backwards_compat():
        """
        Make sure that we can continue to read FITS HDUs that use the schema from
        the ASDF Standard.
    
        This test uses the examples in the fits schema from the ASDF Standard,
        since these make no reference to Astropy's own fits definition.
        """
    
        def check(asdffile):
            assert isinstance(asdffile['example'], fits.HDUList)
    
>       run_schema_example_test('stsci.edu', 'asdf', 'fits/fits', '1.0.0', check)
astropy/io/misc/asdf/tags/fits/tests/test_fits.py:55: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
astropy/io/misc/asdf/tags/tests/helpers.py:14: in run_schema_example_test
    uri = asdf.resolver.default_tag_to_url_mapping(tag)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
uri = 'tag:stsci.edu:asdf/fits/fits-1.0.0'
    def default_tag_to_url_mapping(uri):
>       warnings.warn("'default_tag_to_url_mapping' is deprecated.", AsdfDeprecationWarning)
E       asdf.exceptions.AsdfDeprecationWarning: 'default_tag_to_url_mapping' is deprecated.
.../asdf/resolver.py:169: AsdfDeprecationWarning
____________________________ test_backwards_compat _____________________________
    def test_backwards_compat():
        """
        Make sure that we can continue to read tables that use the schema from
        the ASDF Standard.
    
        This test uses the examples in the table schema from the ASDF Standard,
        since these make no reference to Astropy's own table definition.
        """
    
        def check(asdffile):
            assert isinstance(asdffile['example'], table.Table)
    
>       run_schema_example_test('stsci.edu', 'asdf', 'core/table', '1.0.0', check)
astropy/io/misc/asdf/tags/table/tests/test_table.py:238: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
astropy/io/misc/asdf/tags/tests/helpers.py:14: in run_schema_example_test
    uri = asdf.resolver.default_tag_to_url_mapping(tag)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
uri = 'tag:stsci.edu:asdf/core/table-1.0.0'
    def default_tag_to_url_mapping(uri):
>       warnings.warn("'default_tag_to_url_mapping' is deprecated.", AsdfDeprecationWarning)
E       asdf.exceptions.AsdfDeprecationWarning: 'default_tag_to_url_mapping' is deprecated.
.../asdf/resolver.py:169: AsdfDeprecationWarning
```

Example log: https://travis-ci.org/astropy/astropy/jobs/629110036

As a result, the failures caused unrelated push events to fail as well (e.g., https://github.com/astropy/astropy/pull/9780#issuecomment-568782904).

--- a/astropy/__init__.py
+++ b/astropy/__init__.py
@@ -30,7 +30,7 @@ __minimum_python_version__ = '3.6'
 __minimum_numpy_version__ = '1.16.0'
 # ASDF is an optional dependency, but this is the minimum version that is
 # compatible with Astropy when it is installed.
-__minimum_asdf_version__ = '2.3.0'
+__minimum_asdf_version__ = '2.5.0'
 
 
 class UnsupportedPythonError(Exception):

--- a/astropy/io/misc/asdf/tags/tests/helpers.py
+++ b/astropy/io/misc/asdf/tags/tests/helpers.py
@@ -11,8 +11,8 @@ def run_schema_example_test(organization, standard, name, version, check_func=No
     from asdf.schema import load_schema
 
     tag = format_tag(organization, standard, version, name)
-    uri = asdf.resolver.default_tag_to_url_mapping(tag)
-    r = asdf.AsdfFile().resolver
+    uri = asdf.extension.default_extensions.extension_list.tag_mapping(tag)
+    r = asdf.extension.get_default_resolver()
 
     examples = []
     schema = load_schema(uri, resolver=r)

--- a/setup.cfg
+++ b/setup.cfg
@@ -67,7 +67,7 @@ all =
     matplotlib>=2.1
     scikit-image
     mpmath
-    asdf>=2.3
+    asdf>=2.5
     bottleneck
     ipython
     pytest

